File: python/sort_wstaw.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  sort_wstaw.py
#  


def sort_wstaw(lista):
    """Wersja liniowa"""
    for i in range(1, len(lista)):
        el = lista[i]
        k = i - 1
        while k >= 0 and lista[k] < el:  #  wyszukiwaniepozycji
            lista[k + 1] = lista[k]  #  przesuwanie elementow o 1 do gory
            k = k - 1
        lista[k + 1] = el
    return lista


def main(args):
    lista = [4, 3, 7, 0, 2, 3, 1, 9]
    print(lista)
    print(sort_wstaw(lista))
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/sort_wstaw.cpp
/*
 * sort_wstaw.cpp
 */


#include <iostream>

using namespace std;

void sort_wstaw(int tab[], int n)
{
    int el, k;
    for(int i=1; i<n; i++)
    {
        el = tab[i]; 
        k = i-1;
        while(k>=0 && tab[k]>el) 
        {
            tab[k+1] = tab[k]; 
            k = k - 1;
        }
        tab[k+1] = el; 
     }     
}


int main(int argc, char **argv)
{
    int tab[];
    tab[0]= 4; 
    tab[1]= 3; 
    tab[2]= 7; 
    tab[3]= 0; 
    tab[4]= 2; 
    tab[5]= 3; 
    tab[6]= 1; 
    tab[7]= 9; 
    

    
    cout<<tab[i]<<;
    
    return 0;
}
